language/vm/src/file_format.rs

```diff
@@ -730,7 +728,12 @@ impl SignatureToken {
             ),
             Reference(ty) => Reference(Box::new(ty.substitute(tys))),
             MutableReference(ty) => MutableReference(Box::new(ty.substitute(tys))),
-            TypeParameter(idx) => tys[*idx as usize].clone(),
+            TypeParameter(idx) => {
+                // Assume that the caller has previously parsed and verified the structure of the
+                // file and that this guarantees that type parameter indices are always in bounds.
+                assume!((*idx as usize) < tys.len());
```

Comment:
can you share what `assume! `would translate into?

Reply:
It translates to nothing. It just makes an implicit assumption explicit. MIRAI questions implicit assumptions but stays silent on explicit assumptions.